adi/ad4630.py

```diff
@@ -0,0 +1,231 @@
+#       of one or more patent holders.  This license does not release you
+#       from the requirement that you obtain separate licenses from these
+#       patent holders to use this software.
+#     - Use of the software either in source or binary form, must be run
+#       on or directly connected to an Analog Devices Inc. component.
+#
+# THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+# INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT, MERCHANTABILITY AND FITNESS FOR A
+# PARTICULAR PURPOSE ARE DISCLAIMED.
+#
+# IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, INTELLECTUAL PROPERTY
+# RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+import numpy as np
+from adi.attribute import attribute
+from adi.context_manager import context_manager
+from adi.rx_tx import rx
+
+
+def _sign_extend(value, nbits):
+    sign_bit = 1 << (nbits - 1)
+    return (value & (sign_bit - 1)) - (value & sign_bit)
+
+
+def _bitmask(nbits):
+    mask = 0
+    for i in range(0, nbits):
+        mask = (mask << 1) | 1
+    return mask
+
+
+class ad4630(rx, context_manager, attribute):
+
+    """ AD4630 is low power 24-bit precision SAR ADC """
+
+    _complex_data = False
+    _data_type = "voltage"
+    _device_name = ""
+    _rx_channel_names = []
+    _rx_data_type = np.uint32
+
+    """ Default part to initialize is ad4630-24. If you don't hardware test fails"""
+
+    def __init__(self, uri="", device_name="ad4630-24"):
+
+        context_manager.__init__(self, uri, self._device_name)
+
+        compatible_parts = ["ad4630-24", "ad4030-24", "ad4630-16"]
+
+        if device_name not in compatible_parts:
+            raise Exception(
+                "Not a compatible device: "
+                + device_name
+                + ". Please select from "
+                + str(compatible_parts)
+            )
+        else:
+            self._ctrl = self._ctx.find_device(device_name)
+            self._rxadc = self._ctx.find_device(device_name)
+
+        _channels = []
+        self.output_bits = []
+        for ch in self._ctrl.channels:
+            self.output_bits.append(ch.data_format.bits)
+            self._rx_channel_names.append(ch.id)
+            if "differential" in ch.name:
+                _channels.append((ch.id, self._diff_channel(self._ctrl, ch.id)))
+                if "0" in ch.name:
+                    self.chan0 = self._diff_channel(self._ctrl, ch.name)
+                if "1" in ch.name:
+                    self.chan1 = self._diff_channel(self._ctrl, ch.name)
+                continue

```

Comment:
Unnecessary continue

Reply:
Sorry. I will remove it along with any other changes required to this file. If not I can do it.